elasticdl/pkg/ps/server.go

```diff
@@ -23,11 +23,17 @@ type Server struct {
 }
 
 // NewServer creates a Server instance
-func NewServer(ID int, opt string, lr float32) *Server {
-	return &Server{
-		Param: NewParameter(),
-		Opt:   NewOptimizer(opt, lr),
-		ID:    ID}
+func NewServer(ID int, optType string, optArgs string) *Server {
+	var ps Server
+	ps.Param = NewParameter()
+	var err error
+	ps.Opt, err = NewOptimizer(optType, optArgs)
+	if err != nil {
+		log.Fatalf("failed to create PS server: %v", err)
+	}
+	ps.ID = ID
+	ps.lock = sync.Mutex{}
```

Comment:
We could remove this line. The zero value for a Mutex is an unlocked mutex. Please refer to https://stackoverflow.com/questions/45744165/do-mutexes-need-initialization-in-go

Reply:
Done